### DISCUSSION TITLE
Updating from diffrent nightly versions

### DISCUSSION THREAD

**Discussion by poggersuni** [ikemen-engine/Ikemen-GO#2138] [Q&A] [answered]
How do I update from a previous nightly build to the latest? I've gotten conflicting answers regarding this.

**Comment by UrsoManso** [comment]
Most of the time all you gotta do is copy the external folder and the executable from the latest nightly to your current Ikemen folder.

**Reply by poggersuni** [reply]
I do that and i get the attached error when loading into a match (log and associated error image attached)

![image](https://github.com/user-attachments/assets/849ddf92-0372-4a4e-9d96-0d445ba3d96d)

`.\external\script\start.lua:1422: data/common1.cns.zss:995:
1, ifElse(isHelper, palNo, drawPalNo)
dest: Invalid data: drawpalno
stack traceback:
	[G]: in function 'game'
	.\external\script\start.lua:1422: in function 'f_game'
	.\external\script\start.lua:1859: in function 'launchFight'
	external/script/default.lua:2: in function <external/script/default.lua:0>
	.\external\script\start.lua:1476: in function 'f'
	external/script/main.lua:3132: in function 'loop'
	external/script/main.lua:3253: in function 'loop'
	external/script/main.lua:4220: in main chunk
	[G]: ?
`

**Reply by UrsoManso** [reply]
Try copying the common1.cns.zss from latest nightly as well.

drawPalNo is a invalid command.

**Reply by poggersuni** [reply]
Nope
![image](https://github.com/user-attachments/assets/1419c52b-bc4a-4ff4-9fbc-75090d4712d3)
```
.\external\script\start.lua:1422: chars/jam_xx/jam_xx/jam_xx.def:
./data/score.zss
data/score.zss:65:
No comparison operator
Token = '[' String = 'const(AttrStandingHyperAttack), const(AttrAerialHyperAttack)]'
[ECID 3]

stack traceback:
	[G]: in function 'game'
	.\external\script\start.lua:1422: in function 'f_game'
	.\external\script\start.lua:1859: in function 'launchFight'
	external/script/default.lua:2: in function <external/script/default.lua:0>
	.\external\script\start.lua:1476: in function 'f'
	external/script/main.lua:3132: in function 'loop'
	external/script/main.lua:3253: in function 'loop'
	external/script/main.lua:4220: in main chunk
	[G]: ?

```
and before asking if its something to do with the characters, they all work perfectly before attempting to update.

at this point, im just going to transfer my screenpack,characters,stages,music and modules to the new version if i can't get something at works.

**Reply by UrsoManso** [reply]
All these errors are coming from common files. You probably had a very very outdated version of Ikemen, try replacing all the common files first.

**Reply by poggersuni** [reply]
did that, exact same error

**Reply by UrsoManso** [reply]
That doesn't make any sense to me. Just to be sure you did what I told you at first and copied the entire external folder and executable from latest nightly, right?

**Reply by poggersuni** [reply, marked as answer]
nvm, i think i found the problem on my own. its cuz the modules I had from my own pack werent up-to-date (for reference, i had POTS's active tag,dash cancel, and shadow assist, or rather the launch versions of em). they didnt gel with the new version, thus causing the crash

**Reply by potsmugen** [reply]
Many things broke yesterday with some code deprecation. I did update my modules right away.
